service/history/api/pauseactivity/api_test.go

```diff
@@ -0,0 +1,144 @@
+		mockShard           *shard.ContextTest
+		mockEventsCache     *events.MockCache
+		mockNamespaceCache  *namespace.MockRegistry
+		mockTaskGenerator   *workflow.MockTaskGenerator
+		mockMutableState    *workflow.MockMutableState
+		mockClusterMetadata *cluster.MockMetadata
+
+		executionInfo *persistencespb.WorkflowExecutionInfo
+
+		validator *api.CommandAttrValidator
+
+		logger log.Logger
+	}
+)
+
+func TestActivityOptionsSuite(t *testing.T) {
+	s := new(pauseActivitySuite)
+	suite.Run(t, s)
+}
+
+func (s *pauseActivitySuite) SetupSuite() {
+}
+
+func (s *pauseActivitySuite) TearDownSuite() {
+}
+
+func (s *pauseActivitySuite) SetupTest() {
+	s.Assertions = require.New(s.T())
+
+	s.controller = gomock.NewController(s.T())
+	s.mockTaskGenerator = workflow.NewMockTaskGenerator(s.controller)
+	s.mockMutableState = workflow.NewMockMutableState(s.controller)
+
+	s.mockShard = shard.NewTestContext(
+		s.controller,
+		&persistencespb.ShardInfo{
+			ShardId: 0,
+			RangeId: 1,
+		},
+		tests.NewDynamicConfig(),
+	)
+
+	s.mockNamespaceCache = s.mockShard.Resource.NamespaceCache
+	s.mockClusterMetadata = s.mockShard.Resource.ClusterMetadata
+	s.mockEventsCache = s.mockShard.MockEventsCache
+	s.mockClusterMetadata.EXPECT().GetCurrentClusterName().Return(cluster.TestCurrentClusterName).AnyTimes()
+	s.mockClusterMetadata.EXPECT().GetClusterID().Return(int64(1)).AnyTimes()
+	s.mockClusterMetadata.EXPECT().IsGlobalNamespaceEnabled().Return(true).AnyTimes()
+	s.mockEventsCache.EXPECT().PutEvent(gomock.Any(), gomock.Any()).AnyTimes()
+
+	s.logger = s.mockShard.GetLogger()
+	s.executionInfo = &persistencespb.WorkflowExecutionInfo{
+		VersionHistories:                 versionhistory.NewVersionHistories(&historyspb.VersionHistory{}),
+		FirstExecutionRunId:              uuid.New(),
+		WorkflowExecutionTimerTaskStatus: workflow.TimerTaskStatusCreated,
+	}
+	s.mockMutableState.EXPECT().GetExecutionInfo().Return(s.executionInfo).AnyTimes()
+	s.mockMutableState.EXPECT().GetCurrentVersion().Return(int64(1)).AnyTimes()
+
+	s.validator = api.NewCommandAttrValidator(
+		s.mockShard.GetNamespaceRegistry(),
+		s.mockShard.GetConfig(),
+		nil,
+	)
+}
+
+func (s *pauseActivitySuite) TearDownTest() {
+	s.controller.Finish()
+	s.mockShard.StopForTest()
+}
+
+func (s *pauseActivitySuite) TestPauseActivityAcceptance() {
+	activityInfo := &persistencespb.ActivityInfo{
+		TaskQueue:  "task_queue_name",
+		ActivityId: "activity_id",
+		ActivityType: &commonpb.ActivityType{
+			Name: "activity_type",
+		},
+	}
+
+	request := &historyservice.PauseActivityRequest{
+		NamespaceId: "NamespaceID",
+	}
+
+	s.mockMutableState.EXPECT().IsWorkflowExecutionRunning().Return(true)
+	s.mockMutableState.EXPECT().GetActivityByActivityID(gomock.Any()).Return(activityInfo, true)
+	s.mockMutableState.EXPECT().UpdatePausedEntitiesSearchAttribute().Return(nil)
+	s.mockMutableState.EXPECT().UpdateActivityWithCallback(gomock.Any(), gomock.Any())
+
+	err := pauseActivity(s.mockMutableState, request)
```

Comment:
check that the activity was actually paused?

Reply:
Also not great practice to test unexported functions. TBH, I'd just do a functional test here, this test doesn't provide much value.

Reply:
Everything should be unit-tested is possible.
We can't test that activity was actually paused, mock mutable state is used.
And yes, there will be functional tests on top of that, just not in this PR.